### PR TITLE
Charm immune persistence by zone

### PR DESCRIPTION
**Pull request by grimmier378**
use the stored immune list instead of test checking every pull for id's

don't store HIGH_LVL though this is only temporary (ideally)

since mobs of the same name and type can still be immune if over a certain lvl but not report that it is to high lvl.

### CODE REVIEW

File: modules/charm.lua [hunk 1]
```diff
@@ -321,10 +340,34 @@ function Module:Render()
 					ImGui.TableNextColumn()
 					ImGui.Text(string.format("%s", data.name))
 					ImGui.TableNextColumn()
+					ImGui.Text(string.format("%s", data.lvl))
```
Comment (brainiac) [inline modules/charm.lua:343]: fyi ImGui.Text has string.format built in, so you can just remove the string.format part.

File: modules/charm.lua [hunk 2]
```diff
@@ -363,7 +426,24 @@ function Module:CharmLvlToHigh(mobLvl)
 end
 
 function Module:IsCharmImmune(mobId)
-	return self.TempSettings.CharmImmune[mobId] ~= nil
+	local mobName = mq.TLO.Spawn(mobId).CleanName() or "Unknown"
+	local mobType = mq.TLO.Spawn(mobId).Body() or "Unknown"
+	local zoneShort = mq.TLO.Zone.ShortName()
+	local mobLvl = mq.TLO.Spawn(mobId).Level() or 0
```
Comment (brainiac) [inline modules/charm.lua:432]: you just did a search for the same mob by id 3 times. need to chill out lol

**Comment by brainiac** [comment]
FYI - You changed tings in some places and completely ignored the same problem in others.

File: modules/charm.lua [hunk 3]
```diff
@@ -292,31 +292,23 @@ function Module:Render()
...
-					if data.duration > 30000 then
-						ImGui.PushStyleColor(ImGuiCol.Text, 0.02, 0.8, 0.02, 1)
-					elseif data.duration > 15000 then
-						ImGui.PushStyleColor(ImGuiCol.Text, 0.8, 0.8, 0.02, 1)
-					else
-						ImGui.PushStyleColor(ImGuiCol.Text, 0.8, 0.02, 0.02, 1)
-					end
-					ImGui.Text(string.format("%s", RGMercUtils.FormatTime(math.max(0, data.duration / 1000))))
-					ImGui.PopStyleColor()
+					ImGui.Text("%s", data.name)
```
Comment (brainiac) [inline modules/charm.lua:307]: `ImGui.Text(data.name)`

**Review by DerpleMQ2 — APPROVED** [review]

**Pull request merged**